### ISSUE TITLE
pass in interface ID to create-and-exercise, exercise-by-key commands

### ISSUE THREAD

**Issue by S11001001** [digital-asset/daml#14056]
Spin off #13920. When #13653 introduces the optional interface ID to these commands, change `CreateAnd.java` and `ByKey.java` to pass in the interface ID as mentioned in TODOs in that file.

Note https://github.com/digital-asset/daml/issues/13653#issuecomment-1142178491 for upstream status.

https://github.com/digital-asset/daml/blob/f8766e5ae0efade4c90fc53908dcddabdc6c2712/language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/CreateAnd.java#L44-L46

https://github.com/digital-asset/daml/blob/f8766e5ae0efade4c90fc53908dcddabdc6c2712/language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/ByKey.java#L43-L44

**Comment by chunlokling-da** [comment]
> When https://github.com/digital-asset/daml/issues/13653 introduces the optional interface ID to these commands,

Does it mean that there will be an optional interface ID appearing in the command proto?

https://github.com/digital-asset/daml/blob/9d5115da0bad39a880d3a077948831f6c9e975cd/ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

cc @ray-roestenburg-da

**Comment by ray-roestenburg-da** [comment]
@chunlokling-da yes I'm expecting a change like that. I think I've seen something in PRs but can't seem to find it right now.

**Comment by ray-roestenburg-da** [comment]
@S11001001 Is create-and-exercise with interface ID blocked, or just exercise-by-key?

**Comment by ray-roestenburg-da** [comment]
Blocked until a `choice_interface_id` field is added to the gRPC service, this was discussed with @remyhaemmerle-da in our Zurich meeting but no further work has been done on this.

**Comment by S11001001** [comment]
@michael-godfrey-da For what it's worth, this issue doesn't address notional *interface keys*; the key here is still a template key, which is still an unambiguous way to select the subject of an exercise-by-key when the choice is defined on an interface. [The feature's absence is known.](https://github.com/digital-asset/daml/issues/13653#issuecomment-1142065563)

*Interface keys* would be a quite more involved feature; I don't know whether they are in the pipeline. Given the problems with key uniqueness already, it wouldn't surprise me if they were found to be intractable.

**Comment by remyhaemmerle-da** [comment]
This is a feature request. This needs to be prioritize by product.
